File: pdp_homework2/src/AProduction.c
#include "AProduction.h"
int chooseRandom3() {

    int random_number = rand() % 10+1;  // 0 veya 1 arasında rastgele bir sayı oluştur

    return random_number;
}

AProduction new_AProduction(){
    srand(time(NULL)); // Rastgelelik için zamanı kullanarak tohumlama yapılır
    AProduction *this = (AProduction*)malloc(sizeof(AProduction));
    this->super = new_Production();
    this->super.foodProduct = &foodProduct;

    return *this;
}

void foodProduct(Colony* colony){


    int num_of_Food = chooseRandom3();

    if(colony->foodStock !=0)
        colony->foodStock +=num_of_Food;





}

File: pdp3java/src/pdp3/AProduction.java
package pdp3;
import java.util.Random;

public class AProduction extends Production {

    private int chooseRandom3() {
        Random random = new Random();
        int random_number = random.nextInt(5) + 1;
        return random_number;
    }

    public AProduction(String name){
        super("AProduction");
    }

    @Override
    public void foodProduct() {
    }

    @Override
    public void foodProduct(Colony colony){

        int num_of_Food =chooseRandom3();

        if(colony.getFoodStock() !=0)
            colony.increaseFood(num_of_Food);

    }


}
